### PR TITLE
update specs API for inets application

### PR DESCRIPTION
**Pull request by Whaileee**

### CODE REVIEW

File: lib/inets/src/http_server/httpd.erl [hunk 1]
```diff
@@ -837,6 +839,7 @@ The fields of record `mod` have the following meaning:
 %%%========================================================================
 -type property() :: atom().
 -type ets_table() :: ets:tid().
+-type socket_type() :: ip_comm | ssl.
```
Comment (IngelaAndin) [inline lib/inets/src/http_server/httpd.erl:842]: I am not sure if we ought to have this alias for the ets:tid type although it has a better name! We can take it up one next scrum meeting. What do you think @erlang/team-ps

File: lib/inets/src/http_server/mod_auth.erl [hunk 2]
```diff
@@ -166,6 +169,16 @@ remove(ConfigDB) ->
     ok.
 
 -doc(#{equiv => add_user/6}).
+-spec add_user(UserName, Options) -> true | {error, Reason} when
+      UserName :: string(),
+      Options :: [{password,Password} | {userData,UserData} | {port,Port} | {addr,Address} | {dir,Directory} | {authPassword,AuthPassword}],
+      Password :: string(),
+      UserData :: term(),
+      Port :: non_neg_integer(),
```
Comment (IngelaAndin) [inline lib/inets/src/http_server/mod_auth.erl:177]: We should use inet:port_number().

Comment (IngelaAndin) [reply lib/inets/src/http_server/mod_auth.erl:177]: Can be found in more than one place!

File: lib/inets/src/http_server/mod_auth.erl [hunk 3]
```diff
@@ -194,13 +214,30 @@ If an error occurs, `{error,Reason}` is returned. When
 [`add_user/2`](`add_user/2`) is called, options `Password`, `UserData`, `Port`,
 and `Dir` are mandatory.
 """.
+-spec add_user(UserName, Password, UserData, Address, Port, Directory) -> true | {error, Reason} when
+      UserName :: string(),
+      Password :: string(),
+      UserData :: term(),
+      Port :: non_neg_integer(),
+      Address :: inet:ip4_address() | string() | undefined,
```
Comment (IngelaAndin) [inline lib/inets/src/http_server/mod_auth.erl:222]: Address should be able to be an ip_v6 address too!?

File: lib/inets/src/http_server/httpd_util.erl [hunk 4]
```diff
@@ -546,6 +605,10 @@ from left-to-right on the hunt for longest possible `Path` being a file or a
 directory. Everything after the longest possible `Path`, isolated with a `/`, is
 regarded as `PathInfo`
 """.
+-spec split_path(RequestLine) -> {Path,QueryStringOrPathInfo} when
```
Comment (IngelaAndin) [inline lib/inets/src/http_server/httpd_util.erl:608]: Should this not be an URI  (as the variable) or maybe URIString, a RequestLine would be the whole "GET URI Version" thing!

File: lib/inets/src/inets_app/inets.erl [hunk 5]
```diff
@@ -110,6 +116,14 @@ start(Type) ->
 %% top supervisor.
 %% --------------------------------------------------------------------
 -doc(#{equiv => start/3}).
+-spec start(Service, ServiceConfig) -> Result when
```
Comment (IngelaAndin) [inline lib/inets/src/inets_app/inets.erl:119]: I think we should add some text about that we consider the stand-alone mode deprecated. It is bad!

Comment (Whaileee) [reply lib/inets/src/inets_app/inets.erl:119]: Like doc text or deprecation warning; if it's not already in place? What do you suggest?

File: lib/inets/src/http_server/httpd.erl (hunk 1 above)
Comment (garazdawi) [reply lib/inets/src/http_server/httpd.erl:842]: The teams have never been used for this purpose. We can discuss doing that, but right now it does not work.

Comment (u3s) [reply lib/inets/src/http_server/httpd.erl:842]: what is the intended use for `erlang/team-ps`?

**Comment by garazdawi** [comment]
Make sure to remove this line before you merge: https://github.com/erlang/otp/blob/OTP-27.0-rc3/lib/inets/src/inets_app/inets.mk#L40

**Pull request merged**